pulp_container/app/access_policy.py

```diff
@@ -43,3 +46,101 @@ def has_namespace_obj_perms(self, request, view, action, permission):
+
+        Returns:
+            True if the user has the Permission named by the ``permission`` argument at the model-level.
+                False otherwise.
+        """
+        return request.user.has_perm(permission)
+
+    def has_repoversion_repo_obj_perms(self, request, view, action, permission):
+        """
+        Checks if the current user has object-level permission on the ``repository`` object.
+
+        Args:
+            request (rest_framework.request.Request): The request being made.
+            view (subclass rest_framework.viewsets.GenericViewSet): The view being checked for
+                authorization.
+            action (str): The action being performed, e.g. "destroy".
+            permission (str): The name of the Permission to be checked. In the form
+                `app_label.codename`, e.g. "core.delete_task".
+
+        Returns:
+            True if the user has the Permission named by the ``permission`` argument on the
+            ``repository`` parameter at the object-level. False otherwise.
+        """
+        repo = view.get_object().repository
+        return request.user.has_perm(permission, repo)
+
+    def has_repoversion_repo_model_or_obj_perms(self, request, view, action, permission):
+        """
+        Checks if the current user has model-level or object-level permissions on the ``repository``.
+
+        Args:
+            request (rest_framework.request.Request): The request being made.
+            view (subclass rest_framework.viewsets.GenericViewSet): The view being checked for
+                authorization.
+            action (str): The action being performed, e.g. "destroy".
+            permission (str): The name of the Permission to be checked. In the form
+                `app_label.codename`, e.g. "core.delete_task".
+
+        Returns:
+            True if the user has the Permission named by the ``permission`` at the model-level or on the
+                argument on the ``repository`` parameter at the object-level. False otherwise.
+        """
+        if self.has_model_perms(request, view, action, permission):
+            return True
+        if self.has_repoversion_repo_obj_perms(request, view, action, permission):
+            return True
+        return False
+
+    def has_pushrepoversion_delete_perms(self, request, view, action):
+        """
+        Checks if the current user has permission to remove a repo version of a push repository.
+
+        Args:
+            request (rest_framework.request.Request): The request being made.
+            view (subclass rest_framework.viewsets.GenericViewSet): The view being checked for
+                authorization.
+            action (str): The action being performed, e.g. "destroy".
+        """
+        sufficient_model_perms = (
+            "container.delete_containernamespace",
+            "container.manage_namespace_distributions",
+            "container.delete_containerdistribution",
```

Comment:
I check these model ones and then the same object ones. Did I get the right set of perms? Any of them is enough, right?

Reply:
maybe i am wrong, but should not we just require model or obj perms of the repo,same as with the regular container repo?
We just need to make sure that the created Groups during push operation contains correct set of permissions assigned to each group(owner, collaborator, consumer)

Reply:
See my other comment https://github.com/pulp/pulp_container/pull/212/files#r563954354
We currently do not use delete permissions on push repos at all. We can start using them but it's the reason I started checking distributions and namespaces.